tests/directives/test_item_directive.py

```
@@ -161,3 +164,316 @@ def test_item_node_replacement(self):
+
+        # Verify callback was called
+        callback_mock.assert_called_once()
+
+    def test_add_relation_with_invalid_relation_reports_warning(self):
+        """Test that adding invalid relations triggers warnings"""
+        # Don't add the relation to the collection
+        self.directive.options['unknown_relation'] = 'OTHER_ITEM'
+
+        with patch.object(self.directive, 'get_source_info', return_value=('test.rst', 10)):
+            # This should not crash but should handle gracefully
+            result = self.directive.run()
```

Flake8 (7.3.0)

[error] local variable 'result' is assigned to but never used

(F841)